projects/igniteui-angular/src/lib/list/list-item.component.ts

```diff
@@ -165,7 +170,8 @@ export class IgxListItemComponent implements IListChild {
      */
     @HostListener('click', ['$event'])
     clicked(evt) {
-        this.list.onItemClicked.emit({ item: this, event: evt });
+        this.list.onItemClicked.emit({ item: this, direction: this.lastPanDir, event: evt });
```

Comment:
I would move the direction after the event so we don't break existing logic.
We should also add at least one test covering this change.

Reply:
I reordered the arguments and added 4 new tests.